P0009/P0009.md

```diff
@@ -1407,26 +1418,39 @@ template<size_t... OtherExtents>
 
 ```c++
 template<class... SizeTypes>
-  explicit constexpr extents(SizeTypes... dynamic) noexcept;
+  explicit constexpr extents(SizeTypes... exts) noexcept;
 ```
 
-* [4]{.pnum} *Constraints:*
+* [5]{.pnum} *Constraints:*
     
-    * [4.1]{.pnum} `(is_convertible_v<SizeTypes, size_type> && ...)` is `true`, and
+    * [5.1]{.pnum} `(is_convertible_v<SizeTypes, size_type> && ...)` is `true`, and
     
-    * [4.2]{.pnum} `sizeof...(SizeTypes) == rank_dynamic()` is `true`.
+    * [5.2]{.pnum} `sizeof...(SizeTypes)` equals either `dynamic_rank()` or `rank()`.
 
-* [5]{.pnum} *Effects:*  Initializes `dynamic_extents_` with `dynamic...`.
+* [5]{.pnum} *Preconditions:* If `sizeof...(SizeTypes) == rank()` is `true`,
```

Comment:
This preconditions needs to be always checked. Imagine E = extends<dyn, 1, 2, dyn, 3, 4>. Then E e(1, 1, 2, 4) requires second and third parameter to be equal to static extends. I.e. static_extends may appaer before dynamic extend.

Reply:
Hey, so we had an author meeting and there was pretty strong opposition to the "leave trailing arguments off" option. So we only allow either provide only the dynamic extends or provide all. If you provide only the dynamic extents there is nothing to check.

Reply:
Oh and we can certainly see if LEWG on the 22nd wants to vote on requiring us to do one of the other options) (hopefully while saying it can be forwarded with that change, if the vote outcome is to do something else than what we do now).

Reply:
Ok, thanks. This also answers my other comment.